### PR TITLE
Allow port forwad to different resources

### PR DESCRIPTION
**Pull request by fgimenez**
This will allow us to portforward to services, no need to know the specific pod name.

### CODE REVIEW

File: tunnel.go [hunk 1]
```diff
@@ -16,29 +17,37 @@ import (
...
-	return &Tunnel{
-		config:    config,
-		client:    client,
-		Namespace: namespace,
-		PodName:   podName,
-		Remote:    remote,
-		stopChan:  make(chan struct{}, 1),
-		readyChan: make(chan struct{}, 1),
-		Out:       ioutil.Discard,
+func NewTunnel(client rest.Interface, config *rest.Config, namespace, resource string, remote int) (*Tunnel, error) {
```
Comment (xh3b4sd) [inline tunnel.go]: When I see this I think it would be cool to actually have a proper config struct.

File: tunnel.go [hunk 2]
```diff
@@ -16,29 +17,37 @@ import (
...
+	// resource includes type and name of the resource, like svc/cnr-server or pod/mypod-3445kdfg
```
Comment (xh3b4sd) [inline tunnel.go]: This would be useful as godoc where it is configured so one knows how to set the configuration properly.

Comment (fgimenez) [reply tunnel.go]: Sure, i'll add it together with the config.

File: tunnel.go (hunk 1 above)
Comment (fgimenez) [reply tunnel.go]: Sure, done, I've taken the opportunity to add more initial checks, PTAL.

**Review by xh3b4sd — APPROVED** [review]
Nice.

**Pull request merged**